openapi-cli/src/test/java/io/ballerina/openapi/cmd/BallerinaCodeGeneratorLicenseTests.java

```diff
@@ -0,0 +1,191 @@
+            }
+        } else {
+            Assert.fail("Code generation failed. : " + readOutput(true));
+        }
+    }
+
+    @Test(description = "Test openapi to ballerina generation with default license headers")
+    public void testBothClientServiceGeneration() throws IOException {
+        Path petstoreYaml = resourceDir.resolve(Paths.get("petstore.yaml"));
+        String[] args = {"--input", petstoreYaml.toString(), "-o", this.tmpDir.toString()};
+        OpenApiCmd cmd = new OpenApiCmd(printStream, tmpDir, false);
+        new CommandLine(cmd).parseArgs(args);
+        cmd.execute();
+
+        if (Files.exists(this.tmpDir.resolve("client.bal")) &&
+                Files.exists(this.tmpDir.resolve("utils.bal")) &&
+                Files.exists(this.tmpDir.resolve("types.bal")) &&
+                Files.exists(this.tmpDir.resolve("petstore_service.bal"))) {
+            try {
+                compareFiles("types_for_both_service_client_generations.bal", "types.bal");
+                compareFiles("client.bal", "client.bal");
+                compareFiles("utils_for_both_service_client_generation.bal", "utils.bal");
+                compareFiles("service.bal", "petstore_service.bal");
+                deleteGeneratedFiles(false);
+            } catch (IOException e) {
+                Assert.fail(e.getMessage());
+            }
+        } else {
+            Assert.fail("Code generation failed. : " + readOutput(true));
+        }
+    }
+
+    @Test(description = "Test openapi to ballerina generation with license headers")
+    public void testUserGivenLicenseHeader() throws IOException {
+        Path petstoreYaml = resourceDir.resolve(Paths.get("petstore.yaml"));
+        Path licenseHeader = resourceDir.resolve(Paths.get("expected_gen/licenses/license.txt"));
+        String[] args = {"--input", petstoreYaml.toString(), "-o", this.tmpDir.toString(), "--license",
+                licenseHeader.toString(), "--with-service-type"};
+        OpenApiCmd cmd = new OpenApiCmd(printStream, tmpDir, false);
+        new CommandLine(cmd).parseArgs(args);
+        cmd.execute();
+        if (Files.exists(this.tmpDir.resolve("client.bal")) &&
+                Files.exists(this.tmpDir.resolve("utils.bal")) &&
+                Files.exists(this.tmpDir.resolve("types.bal")) &&
+                Files.exists(this.tmpDir.resolve("service_type.bal")) &&
+                Files.exists(this.tmpDir.resolve("petstore_service.bal"))) {
+            try {
+                compareFiles("schema_with_user_given_license.bal", "types.bal");
+                compareFiles("client_with_user_given_license.bal", "client.bal");
+                compareFiles("utils_for_with_user_given_license.bal", "utils.bal");
+                compareFiles("service_type_with_user_given_license.bal", "service_type.bal");
+                compareFiles("service_with_user_given_license.bal", "petstore_service.bal");
+                deleteGeneratedFiles(false);
```

Comment:
Let's move the cleanup task into a `final` block for guaranteed execution even when there's an exception